### PR TITLE
Update the validation of the privilege property on the windows_user_privilege resource

### PR DESCRIPTION
**Pull request by tas50**
Built off #10421

### CODE REVIEW

File: lib/chef/resource/windows_user_privilege.rb [hunk 1]
```diff
@@ -137,10 +137,20 @@ class WindowsUserPrivilege < Chef::Resource
...
+              these_options = Array[n]
+            else
+              these_options = n
+            end
+
+            if (these_options - PRIVILEGE_OPTS).empty?
+              true
+            else
+              false
+            end
```
Comment (phiggins) [inline lib/chef/resource/windows_user_privilege.rb:151]:
```suggestion
            (Array(n) - PRIVILEGE_OPTS).empty?
```

`Array()` is a function that will convert its parameter to an array. `Array[]` is the same as bare `[]`, that is, it creates an array containing its parameters.

Comment (phiggins) [reply lib/chef/resource/windows_user_privilege.rb:151]: If the `coerce` proc is run first the conversion might not be necessary.

Comment (tas50) [reply lib/chef/resource/windows_user_privilege.rb:151]: You're too fast here @phiggins. I already have those same changes locally. This isn't done

File: lib/chef/resource/windows_user_privilege.rb [hunk 2]
```diff
@@ -137,10 +137,20 @@ class WindowsUserPrivilege < Chef::Resource
         required: true,
         coerce: proc { |v| v.is_a?(String) ? Array[v] : v },
         callbacks: {
-           "Option privilege must include any of the: #{privilege_opts}" => lambda { |v|
-             (privilege_opts & v).size == v.size
-           },
-         }
+          "Option privilege must include any of the: #{PRIVILEGE_OPTS}" => lambda { |n|
```
[the remark this answers is not in the document]
Comment (tas50) [reply lib/chef/resource/windows_user_privilege.rb]: I pulled this in but changed option to property

File: kitchen-tests/cookbooks/end_to_end/recipes/windows.rb [hunk 3]
```diff
@@ -38,6 +38,11 @@
   action :enable
 end
 
+windows_user_privilege "BUILTIN\\Administrators" do
```
Comment (tas50) [inline kitchen-tests/cookbooks/end_to_end/recipes/windows.rb:41]: @chef-davin Am I missing something here. Isn't the name property the principal not the users?

**Comment by phiggins** [comment]
I read #10420 and I'm confused what the actual problem was and how this PR addresses it.

**Review by phiggins — APPROVED** [review]

**Comment by tas50** [comment]
There's more to that issue than this fix, but this resolves some strange code in the validation, gives us users as strings and adds specs

**Pull request merged**